src/Workspaces/Core/Portable/Shared/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -153,14 +153,20 @@ public static ITypeSymbol RemoveNullableIfPresent(this ITypeSymbol symbol)
             // instantiations of that method.
             var originalInterfaceType = interfaceMember.ContainingType.OriginalDefinition;
             var originalInterfaceMember = interfaceMember.OriginalDefinition;
+
             var constructedInterfaces = typeSymbol.AllInterfaces.Where(i =>
                 SymbolEquivalenceComparer.Instance.Equals(i.OriginalDefinition, originalInterfaceType));
 
+            // Try to get the compilation for the symbol we're searching for, 
+            // which can help identify matches with the call to SymbolFinder.OriginalSymbolsMatch
+            var typeSymbolProject = solution.GetProject(typeSymbolAndProjectId.ProjectId);
+            var typeSymbolCompilation = typeSymbolProject?.GetCompilationAsync(cancellationToken).WaitAndGetResult_CanCallOnBackground(cancellationToken);
+
             foreach (var constructedInterface in constructedInterfaces)
             {
                 cancellationToken.ThrowIfCancellationRequested();
                 var constructedInterfaceMember = constructedInterface.GetMembers().FirstOrDefault(m =>
-                    SymbolEquivalenceComparer.Instance.Equals(m.OriginalDefinition, originalInterfaceMember));
```

Comment:
i'm concerned this won't be the only case that needs to be audited.  why not just have SymbolEquivalenceComparer be updated to understand these cases.  Having two ways of asking the exact same thing seems like a bad place to be.

Reply:
Updating `SymbolEquivalenceComparer` is a larger change and less targetted for this fix. In general I would agree if i were adding a new way to compare symbols, but I'm still only utilizing an existing functionality. If the current spread of logic isn't desired, we should file a separate issue and address accordingly.

Reply:
> Updating `SymbolEquivalenceComparer` is a larger change and less targetted for this fix.

I'm skeptical.  The entire purpose of SymbolEquivalenceComparer is for this sort of thing.  So, if we don't fix that issue, we're just kicking the can down the road.  i.e. users with these same projects will likely see other semantic operations fail beyond just Find-Impl.  I would prefer a broader, more correct fix here, rather than the targetted fix.

Reply:
`SymbolFinder.OriginalSymbolsMatch` does end up using `SymbolEquivalenceComparer`, it just has some more heuristics around it. http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/FindSymbols/SymbolFinder_Hierarchy.cs,9defddf5fea8d771 

I guess what I'm saying is I'm not sure that logic belongs in `SymbolEquivalenceComparer`. I don't have the breadth of knowledge to make that call. If it does, I'd prefer we file a task to remove `SymbolFinder.OriginalSymbolsMatch` and exchange to `SymbolEquivalenceComparer.IgnoreAssembliesInstance.Equals`. 

It may be sufficient to have just `SymbolEquivalenceComparer.IgnoreAssembliesInstance.Equals` here, but I opted to follow other places that seemed similar and include the added checks in `SymbolFinder.OriginalSymbolsMatch`.

Reply:
> I don't have the breadth of knowledge to make that call.

I wrote `SymbolEquivalenceComparer`.  And this was def something it was intended to support :D

I guess my poitn is this.  If we are considreing these symbols from different compilations equivalent (which seems to be how the compiler would view things), then we should update our helper to properly encode the right logic that makes it agree with what the compiler would think.  I don't think that's necessarily `IgnoreAssembliesInstance`, however it might be something else.

Do you know *why* it is that `System.Runtime.dll` and `netstandard.dll` are unifying here?  Is there a type-forwarding situation going on?

--

In other words, i'd prefer we deepely understand this and put in the right *full* fix, rather than a one-off patch at one layer that will work for this one specific case, but not be sufficient for other things we may run into later.

Reply:
> Do you know why it is that System.Runtime.dll and netstandard.dll are unifying here? Is there a type-forwarding situation going on?

Yes, this is type-forwarding. For netstandard from netframework (or even netframework) reference binaries are added to do type-forwarding so `netstandard.dll` types can resolve to the equivalent type. I had an easier time viewing this in netframework build outputs. One of the referenced files is `{VsInstallPath}\MSBuild\Microsoft\Microsoft.NET.Build.Extensions\net461\lib\netstandard.dll` which contains all of the type forwarding. I'm not quite sure how it's done with netcore (I didn't look heavily), but I imagine there's something similar. 

> I guess my poitn is this. If we are considreing these symbols from different compilations equivalent (which seems to be how the compiler would view things), then we should update our helper to properly encode the right logic that makes it agree with what the compiler would think. I don't think that's necessarily IgnoreAssembliesInstance, however it might be something else.

This IS that something else. Is it just that it's not part of statics exposed by `SymbolEquivalenceComparer`? 

To me this is just swapping the wrong call to the right one. If the logic isn't in the right place, then I guess moving it would be okay? I don't feel strongly one way or another. 

I think `SymbolEquivalenceComparer` does the advertised thing and works in the expected way; you compare with a requirement for same assemblies or you compare without. I'd even be willing to move this code to calling it directly. Like I said, I just chose an example from something similar and went with it because it looked like `SymbolEquivalenceComparer` + some extra checks to ensure type forwarding is considered.

Reply:
I do wonder if we should just add an extension method for `ISymbol` to facilitate these type of decisions. Looking for the correct static instance is a pain. 

`ISymbol.Equals` => `SymbolEquivalenceComparer.Instance.Equals`
`ISymbol.EqualsIgnoreAssembly` => `SymbolEquivalenceComparer.IgnoreAssembliesInstance`
`ISymbol.EqualsIgnoreAssembly(other, solution, thisCompilation, otherCompilation, cancellationToken)` => `SymbolFinder.OriginalSymbolsMatch`

Reply:
> Yes, this is type-forwarding.

We really should just be makign SymbolEquivalenceComparer understand type forwarding.  If ti's a forwarded type, then they're equivalent.

> Is it just that it's not part of statics exposed by `SymbolEquivalenceComparer`?

Right.  The statics that are there should be updaed to understand type forwarding. 

> I think `SymbolEquivalenceComparer` does the advertised thing and works in the expected way;

I disagree.  A symbol forwarded to anotehr assembly should be equivalent to the original symbol.  That's what 'equivalence' is supposed to mean here.  i.e. "does hte compiler consider these symbols interchangeable?"  In this case, the answer is yes.  We just never added hte logic for type forwarding here since we didn't need to and there weren't reports of a problem.  Now that there are, we should fix htis.